torchci/pages/api/flaky-tests/flakytest.ts

```diff
@@ -1,50 +1,113 @@
 import type { NextApiRequest, NextApiResponse } from "next";
-import fetchFlakyTests from "lib/fetchFlakyTests";
-import { FlakyTestData, JobData } from "lib/types";
-import fetchFailureSamples from "lib/fetchFailureSamples";
-
-interface Data {}
-
-// The captures are based on the regex in the Python flaky unittest
-// classifications in aws/lambda/log-classifier/ruleset.toml
-export function getFlakyTestCapture(flakyTest: FlakyTestData): string {
-  return `${flakyTest.name}`;
-}
+import { JobData } from "lib/types";
+import getRocksetClient from "lib/rockset";
+import _ from "lodash";
 
 export default async function handler(
   req: NextApiRequest,
-  res: NextApiResponse<Data>
+  res: NextApiResponse
 ) {
-  const name = req.query.name;
-  const suite = req.query.suite;
-  const file = req.query.file;
+  const name = req.query.name as string;
+  const suite = req.query.suite as string;
+  const file = req.query.file as string;
 
-  let numHours = 30 * 24 + "";
-
-  const flakyTests: FlakyTestData[] = await fetchFlakyTests(
-    numHours,
-    name as string,
-    suite as string,
-    file as string
-  );
+  res.status(200).json(await getFlakyTestInfo(name, suite, file));
+}
 
-  // capture looks like: testName, testSuite
-  const flakySamples: {
-    [capture: string]: JobData[];
-  } = {};
+export interface FlakyTestInfoHUD {
+  name: string;
+  classname: string;
+  file: string;
+  invoking_file: string;
+  jobs: JobData[];
+}
 
-  // only get log view for at most 5 flaky tests (to not spam query calls)
-  console.debug(`Retrieved ${flakyTests.length} flaky tests`);
-  if (flakyTests.length <= 5) {
-    const unfulfilledPromises = flakyTests.map(function (flakyTest) {
-      return fetchFailureSamples(getFlakyTestCapture(flakyTest));
-    });
+async function getFlakyTestInfo(
+  name: string,
+  suite: string,
+  file: string
+): Promise<FlakyTestInfoHUD[]> {
+  const query = `
+  select
+  t.name,
+  t.classname,
+  t.file,
+  t.invoking_file,
+  j.conclusion,
+  j.id as job_id,
+  j.name as job_name,
+  j.html_url as job_url,
+  j.started_at as job_started_at,
+  j.torchci_classification.line,
+  j.torchci_classification.line_num,
+  j.torchci_classification.captures
+from
+  workflow_job j
+  join commons.failed_tests_run t on j.id = t.job_id HINT(join_strategy = lookup)
+where
+  t.name like :name
+  and t.classname like :suite
+  and t.file like :file
+  and j.name not like '%rerun_disabled_tests%'
+order by
+  PARSE_TIMESTAMP_ISO8601(j.started_at) desc
+limit
+  100
```

Comment:
I think you could make this limit a param, i.e. https://torchci-git-csl-flakytestspage-fbopensource.vercel.app/flakytest?name=test_dynamic_stride_nobreak_cuda&suite=TestInductorDynamicCUDA **&limit=1000**, which could give some flexibility here.